pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -499,36 +516,36 @@ def _format_checkpoint_name(
 
         return filename
 
-    def format_checkpoint_name(self, epoch: int, step: int, metrics: Dict[str, Any], ver: Optional[int] = None) -> str:
+    def format_checkpoint_name(self, metrics: Dict[str, _METRIC], ver: Optional[int] = None) -> str:
```

Comment:
the few users who subclass model checkpoint will have to change this too. But it's unavoidable I guess.

Reply:
Instead of deprecating, you could have it listed in the changelog.

Reply:
Good point